Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift

```diff
@@ -201,22 +268,38 @@ public final class ViewAnnotationManager {
             validate(view)
 
             view.translatesAutoresizingMaskIntoConstraints = true
-            view.frame = CGRect(
-                origin: position.leftTopCoordinate.point,
-                size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
-            )
+            let oldFrame = view.frame
+            let newFrame = position.frame
+
+            view.frame = newFrame
+
+            if oldFrame != view.frame {
+                viewsWithUpdatedFrame.insert(view)
+            }
+            if view.isHidden {
+                viewsWithUpdatedVisibility.insert(view)
+            }
             view.isHidden = false
             expectedHiddenByView[view] = false
             visibleAnnotationIds.insert(position.identifier)
         }
 
+        assert(!viewsWithUpdatedFrame.contains(where: {$0.isHidden == true }))
+        notifyViewAnnotationObserversFrameDidChange(for: Array(viewsWithUpdatedFrame))
```

Comment:
Would it make sense to wait until the end of the method just before `notifyViewAnnotationObserversVisibilityDidChange` to make this call?

Reply:
Just thinking it might avoid letting the developer see the views in an inconsistent state.

Reply:
It also may make sense to put this block to the `defer`

Reply:
I think the idea with `defer` will work well. It will make more sense to first notify about visibility changes and only after that is established to notify about frame changes.